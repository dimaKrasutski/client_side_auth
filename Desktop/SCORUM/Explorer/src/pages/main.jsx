import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'ramda';
import { connect } from 'react-redux';
import { ErrorBoundary } from 'theme/components/error-boundary/error-boundary';
import { Header } from 'theme/components/header/header';
import { LogoLink } from 'theme/components/header/logo-link/logo-link';
import { GlobalAlert } from 'theme/components/alerts/global-alert/global-alert';
import { AlertGeneral } from 'theme/components/alerts/alert-general/alert-general';
import { AlertLinks } from 'theme/components/alerts/alert-links/alert-links';
import { IconLogo } from '../components/icons/icon-logo/icon-logo';
import { clearError } from '../actions/system';
import { Menu } from '../components/menu/menu';
import './main.scss';

export const MainUI = ({
  errorMessageId,
  errorLinks,
  children,
  dismissGlobalAlert,
  locale,
}) => (
  <main className="mb-6 ">
    <Header>
      <div>
        <LogoLink to="/" locale={locale}>
          <IconLogo />
        </LogoLink>
        <Menu />
      </div>
    </Header>
    {errorMessageId &&
    <GlobalAlert onDismiss={dismissGlobalAlert}>
      {errorLinks ?
        <AlertLinks
          errorMessageId={errorMessageId}
          errorLinks={errorLinks}
          onClick={dismissGlobalAlert}
          locale={locale}
        /> :
        <AlertGeneral errorMessageId={errorMessageId} />}
    </GlobalAlert>
    }

    <ErrorBoundary onDismissAlert={dismissGlobalAlert}>
      {children}
    </ErrorBoundary>
  </main>
);

MainUI.propTypes = {
  locale: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  dismissGlobalAlert: PropTypes.func.isRequired,
  errorMessageId: PropTypes.string,
  errorLinks: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
    messageId: PropTypes.string.isRequired,
  })),
};

MainUI.defaultProps = {
  errorMessageId: null,
  errorLinks: null,
};

const mapStateToProps = ({ system, intl }) => ({
  errorMessageId: system.globalError.id,
  errorLinks: system.globalError.links,
  locale: intl.localeFull,
});

const mapDispatchToProps = dispatch => ({
  dismissGlobalAlert: compose(dispatch, clearError),
});

export const Main = connect(mapStateToProps, mapDispatchToProps)(MainUI);
