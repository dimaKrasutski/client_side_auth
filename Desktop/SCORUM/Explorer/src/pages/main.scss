.logo-text {
  font-weight: 300;
}

.container { width: 1280px !important; }

.container-fluid {
  min-width: 1280px;
}
