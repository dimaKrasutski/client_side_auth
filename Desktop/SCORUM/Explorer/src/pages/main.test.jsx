import React from 'react';
import { MainUI } from './main';

describe('<MainUI />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<MainUI locale="en-us" dismissGlobalAlert={jest.fn()} >test</MainUI>);
    expect(wrapper).toMatchSnapshot();
  });
});
