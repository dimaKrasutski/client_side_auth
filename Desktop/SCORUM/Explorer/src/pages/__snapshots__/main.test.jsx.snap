// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainUI /> should match snapshot 1`] = `
<main
  className="mb-6 "
>
  <Header>
    <div>
      <LogoLink
        locale="en-us"
        to="/"
      >
        <IconLogo />
      </LogoLink>
      <Menu />
    </div>
  </Header>
  <ErrorBoundary
    message={null}
    onDismissAlert={[MockFunction]}
  >
    test
  </ErrorBoundary>
</main>
`;
